crates/net/network/src/cache.rs

```diff
@@ -9,12 +19,19 @@ use std::{borrow::Borrow, hash::Hash, num::NonZeroUsize};
 pub struct LruCache<T: Hash + Eq> {
     limit: NonZeroUsize,
     inner: LinkedHashSet<T>,
+    tx: Option<UnboundedSender<T>>,
 }
 
 impl<T: Hash + Eq> LruCache<T> {
-    /// Creates a new `LruCache` using the given limit
+    /// Creates a new [`LruCache`] using the given limit
     pub fn new(limit: NonZeroUsize) -> Self {
-        Self { inner: LinkedHashSet::new(), limit }
+        Self { inner: LinkedHashSet::new(), limit, tx: None }
+    }
+
+    /// Creates a new [`LruCache`] using the given limit and with a channel for sending eviction
+    /// feedback. Note, that it is up to the caller's context to drain the buffer!
+    pub fn new_with_feedback(limit: NonZeroUsize, tx: UnboundedSender<T>) -> Self {
```

Comment:
looks like `new_with_feedback` and `remove` will need to be `pub(crate)` to satisfy clippy